keps/sig-scheduling/5598-scheduling-cache/README.md

```diff
@@ -0,0 +1,955 @@
+
+If plugin changes prove to be an issue, we could codify the signature as a new "Scheduling" object that only has a subset
+of the fields of the pod. Plugins that "opt-in" could only be given access to this reduced scheduling object, and we could then use the entire scheduling object as the signature. This would make it more or less impossible for the signature and plugins to be out of sync, and would
+naturally surface new dependencies as additions to the scheduling object. However, as we expect plugin changes to be relatively 
+modest, we don't believe the complexity of making the interface changes is worth the risk today.
+
+#### We are narrowing the feature set where the cache will work
+
+Because we are explicitly limiting the functionality that this cache will support, we run the risk of designing something
+that will not work for enough customers for it to be useful.
+
+To mitigate this risk we are actively engaging with customers and doing analysis of data available on K8s users to ensure we are still
+capturing a large enough number of user use cases. We also will address this by expanding over time; we expect to have a few interested
+parties up front, but will then evaluate expansions that could onboard more.
+
+## Design Details
+
+### Pod signature v1
+
+The follow section outlines the attributes we are currently proposing to use as the signature for each of the 
+plugins in the scheduler. We need the plugin owners to validate that these signatures are correct, or help
+us find the correct signature.
+
+Note that the signature does not need to be stable across versions, or even invocations of the scheduler. 
+It only needs to be comparable between pods on a given running scheduler instance.
+
+ * DynamicResources: For now we mark a pod unsignable if it has dynamic resource claims. We should improve this in the future, since most
+ resource claims will allow for a signature.
+ * ImageLocality: We use the canonicalized image names from the Volumes as the signature.
+ * InterPodAffinity: If either the PodAffinity or PodAntiAffinity fields are set, the pod is marked unsignable, otherwise an empty signature.
+ * NodeAffinity: We use the NodeAffinity and NodeSelector fields as the signature.
+ * NodeName: We use the NodeName field as the signature.
+ * NodePorts: We use the results from util.GetHostPorts(pod) as the signature.
+ * NodeResourcesBalancedAllocation: We use the output of calculatePodResourceRequestList as the signature.
+ * NodeResourcesFit: We use the output of the computePodResourceRequest function as the signature.
+ * NodeUnschedulable: We use the Tolerations field as the signature.
+ * NodeVolumeLimits: We use all Volume information except from Volumes of type ConfigMap or Secret.
+ * PodTopologySpread: If the PodTopologySpead field is set, or it is not set but a default set of rules are applied, we mark the pod unsignable, otherwise it returns an empty signature.
```

Comment:
> a default set of rules are applied, we mark the pod unsignable

Meaning, all pods that belong to Service, ReplicaSet, StatefulSet or ReplicationController are not eligible to use the cache. That's a large portion.